auth.php

```diff
@@ -145,6 +145,26 @@ public function user_signup($user, $notify=true) {
             }
         }
 
+        // Lookup group enrol keys. Not forgetting that group enrolment key is kept in {group}.enrolmentkey.
+        $enrolplugins = $DB->get_records_sql("
+                SELECT e.*, g.enrolmentkey
```

Comment:
We need those fields because later we call $enrol_self_plugin->enrol_self where all those fields from table {enrol} are used.